mne/export/_edf.py

```diff
@@ -206,6 +206,12 @@ def _export_raw(fname, raw, physical_range, add_ch_type):
     # compute number of data records to loop over
     n_blocks = n_times / out_sfreq
 
+    # increase the number of annotation signals if necessary
+    n_annotations = len(raw.annotations)
+    if n_annotations > n_blocks:
+        n_annot_signals = np.ceil(n_annotations / n_blocks).astype(int)
+        hdl.setNumberOfAnnotationSignals(n_annot_signals)
```

Comment:
this will request 200 annotation channels if there are 200 annotations.  That seems... suboptimal.  If the Raw file is 600 seconds (10 minutes) long and has a 0.5-second annotation every 3 seconds, you would have 200 annotations and yet could write all of them on a single annotation channel (because they're guaranteed to be in different data blocks).

I don't know the ins and outs of how EDFLib handles the annotations (e.g., does a single annotation persist across data blocks until its offset, or are only the onsets and durations written in a given block?) but it seems like comparing the annotation onset times (and maybe durations) to the data blocks' start and stop times should allow you to efficiently figure out how many channels you will need.  `np.searchsorted` springs to mind, though there may be more elegant ways.

Reply:
Neither the onset time or duration of an annotation affects the number of annotations that can be written.  
EDFlib works basically like this: you can write as many annotations as you like using the api. They will  
be temporary stored in memory. Only when you close the file, EDFlib will set the filepointer to the first  
datarecord and write an annotation, then hops to the next datarecord and write the next annotation and  
so on untill there are no annotations or datarecords anymore, whichever one comes first.  
  
If the number of annotations and the number of datarecords are known before writing the datasamples,  
the following formula can be used to calculate the number of annotation channels needed:  
  
`annot_channs = (annots / datrecs) + (annots % datrecs);`  
`if(annot_channs > 1)`  
`{`  
`  setNumberOfAnnotationSignals(annot_channs);`  
`}`

Reply:
Thanks for the explanation @Teuniz! 

@drammock I've adjusted the code to address your comments.

Reply:
Ah ok, I was wrong, I see it works very differently than I assumed. Thanks @Teuniz for the explanation. @adam2392 I'll review again first thing tomorrow and hopefully get this merged!